templates/pkg/resource/references.go.tpl

```diff
@@ -18,10 +18,28 @@ import (
 	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
 {{ end -}}
 	acktypes "github.com/aws-controllers-k8s/runtime/pkg/types"
-
 	svcapitypes "github.com/aws-controllers-k8s/{{ .ServicePackageName }}-controller/apis/{{ .APIVersion }}"
+{{ $servicePackageName := .ServicePackageName -}}
+{{ $apiVersion := .APIVersion -}}
+{{ if .CRD.HasReferenceFields -}}
+{{ range $fieldName, $field := .CRD.Fields -}}
+{{ if and $field.HasReference (not (eq $field.ReferencedServiceName $servicePackageName)) -}}
+    {{ $field.ReferencedServiceName }}apitypes "github.com/aws-controllers-k8s/{{ $field.ReferencedServiceName }}-controller/apis/{{ $apiVersion }}"
```

Comment:
Here you are using the "Service package name", no the same as the "Service model name" that you ask for in the comments of the generator `ReferencesConfig`.

Reply:
Actually, I do need the ServicePackageName i.e. "Used by the ACK controller's for their name"